tests/integration/pilot/vm/README.md

```diff
@@ -0,0 +1,88 @@
+to utilize actual compute instances from different providers.**
+
+## Overview
+
+This section describes the scenarios in which one might want to add a VM test:
+1. Testing existing core Istio features such as traffic management, security, telemetry, etc. for **VMs**
+2. Supporting new OS images for VMs
+3. Testing onboarding tools (iptables, certs, istio-sidecar, etc.) and workflows (services, DNS, etc.) to enmesh a VM
+
+## Secenario 1: Testing VM-related Istio Code
+
+We normally test Istio using Echo application. To test VMs' connectivity, security and telemetry,
+we deploy a regular Echo instance as VM. A VM Echo instance will simulate a VM, disabling kube-dns,
+Service Account mount, etc. For more information around VM onboarding,
+refer to this [doc](https://istio.io/latest/docs/examples/virtual-machines/single-network/).
+
+To deploy an echo instance as a VM
+1. Set the ports for the VMs. Note that due to a bug in WorkloadEntry,
+we need to set the ServicePort to be the same as the InstancePort
+2. Set `DeployAsVm` to be true in `echo.Config`
+3. Set `VMImage` to be any image available from `GetSupportedOSVersion` in [util.go](https://github.com/istio/istio/blob/master/tests/integration/pilot/vm/util.go).
+
+For example,
+
+```go
+ports := []echo.Port{
+   {
+       Name:     "http",
+       Protocol: protocol.HTTP,
+       // Due to a bug in WorkloadEntry, service port must equal target port for now
+       InstancePort: 8090,
+       ServicePort:  8090,
+   },
+}
+echo.Config{
+   Service:    "vm",
+   Namespace:  "virtual-machine",
+   Ports:      ports,
+   Pilot:      p,
+   DeployAsVM: true,
+   VMImage:    "app_sidecar_ubuntu_bionic"
+}
+```
+
+There is a default image and one could reference it with `DefaultVMImage` so that it's easier to maintain
+if the supported OSes were to change. We have [additional images](https://github.com/istio/istio/blob/master/tests/integration/pilot/vm/util.go)
+that are built in the post-submit jobs, jobs that only run after a PR is merged.
+Info regarding adding extra images could be found in the next section.
+If `VMImage` is not provided while `DeployAsVM` is on, it will default the Docker image to be `DefaultVMImage`.
+
+## Scenario 2: Supporting More OS Images
+
+We list the supported OSes in [util.go](https://github.com/istio/istio/blob/master/tests/integration/pilot/vm/util.go)
+and the images will be created in [prow/lib.sh](https://github.com/istio/istio/blob/master/prow/lib.sh).
+
+To add additional supported images for testing:
+1. Modify [tools/istio-docker.mk](https://github.com/istio/istio/blob/master/tools/istio-docker.mk) to add more
+build targets. See other build targets for references. Basically, one needs to specify the OS image name and version,
+and pass it to the Dockerfile.
+2. Modify `prow/lib.sh` by adding images to the targets so that the image could be built on fly for CI/CD.
```

Comment:
Instead of 'so that the image could be build on fly' maybe 'to build the image for CI/CD.'